### PR TITLE
Copy `cortex/pkg/querier/astmapper` package dependency into Loki

### PR DESCRIPTION
**Pull request by DylanGuedes**
**What this PR does / why we need it**:
As a part of removing Loki dependency on cortex packages, we decided to copy the `querier/astmapper` package from cortex to Loki itself.
I decided to only fork this package and not whole querier because the other subpackages require code changes and the gigantique diff will make reviews a hard task. For this one I just had to copy stuff and replace `cortex/astmapper` with the new counterpart. 

**Which issue(s) this PR fixes**:
NA

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by trevorwhitney — APPROVED** [review]
LGTM. One quesiton about the `util_logs` package, are we waiting to port those all off cortex at the same time, because I noticed one came over with the astmapper package.

File: pkg/querier/astmapper/parallel.go
Comment (trevorwhitney) [inline pkg/querier/astmapper/parallel.go:9]: should this still be pointing to cortex?

**Comment by DylanGuedes** [comment]
> LGTM. One quesiton about the `util_logs` package, are we waiting to port those all off cortex at the same time, because I noticed one came over with the astmapper package.

Hmm that's a good question. There's already an open PR that get rid of Cortex util/util_logs, but since it is quite big and I'm expecting that it will take some time to get merged, I decided to send this one for review as it might be easier to review/merge.
That said, I'm fine with waiting for the `util` fork pkg to be merged before this one but I'm also fine with merging this one first..

File: pkg/querier/astmapper/parallel.go
Comment (DylanGuedes) [reply pkg/querier/astmapper/parallel.go:9]: yes, until [this PR](https://github.com/grafana/loki/pull/4917) gets merged.

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**
